_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -944,6 +944,23 @@ mfxStatus ImplementationAvc::Init(mfxVideoParam * par)
         }
     }
     mfxExtCodingOption3 & extOpt3 = GetExtBufferRef(m_video);
+
+    if (extOpt2.IntRefCycleSize == 1)
```

Comment:
Could you clarify why do you check IntRefCycleSize out of verification of IntRefType?

I suggest to put this verification as part of IntRefType conidition https://github.com/Intel-Media-SDK/MediaSDK/blob/4e98b9c1355d0f81fbefc75f12542aeed8c63be5/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp#L1166

According documentation 0 is also wrong value (actually, in case of MFX_REFRESH_SLICE it will be ignored). And it should be explicitly verified because otherwise we have division by zero at https://github.com/Intel-Media-SDK/MediaSDK/blob/4e98b9c1355d0f81fbefc75f12542aeed8c63be5/_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp#L1175

Also I'm wondering why I cannot find line where we check FrameRateExtD!=0 (for filling default value in case of IntRefType!=MFX_REFRESH_SLICE)...

Reply:
I moved verification inside `IntRefType `condition as you suggested. 
0 value already being processed inside `SetDefaults `here
https://github.com/Intel-Media-SDK/MediaSDK/blob/7e3ef73f9b4503f68721d9b329536bf3d119dddb/_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp#L5925
It called from `CheckVideoParam` here
https://github.com/Intel-Media-SDK/MediaSDK/blob/7e3ef73f9b4503f68721d9b329536bf3d119dddb/_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp#L2067
And  `CheckVideoParam` is called before verification.
As for `FrameRateExtD!=0` it verified inside `CheckVideoParam` here 
https://github.com/Intel-Media-SDK/MediaSDK/blob/7e3ef73f9b4503f68721d9b329536bf3d119dddb/_studio/mfx_lib/shared/src/mfx_h264_enc_common_hw.cpp#L1964-L1966